### ISSUE TITLE
ImageDataGenerator is not shuffle during training when shuffle = True

### ISSUE THREAD

**Issue by xorangecheng** [keras-team/keras#10166]
Hi,
In the image.py, the on_epoch_end should be called after each epoch to make sure the data is shuffled for next epoch. But it's not called, that means the data only shuffle once at the beginning of the training.

**Comment by Dref360** [comment]
I guess you talk about the keras.preprocessing.image.Iterator object returned by `.flow`.


The Iterator is a `keras.utils.Sequence` which implements the `on_epoch_end` callback.

**Comment by xorangecheng** [comment]
Hi Dref360, thanks for your reply. I just print the index_array of each batch, it is actually shuffled.
